src/main/java/com/example/bookshop/service/implementation/CategoryServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.example.bookshop.service.implementation;
+
+import com.example.bookshop.dto.CategoryDto;
+import com.example.bookshop.exception.EntityNotFoundException;
+import com.example.bookshop.mapper.CategoryMapper;
+import com.example.bookshop.model.Category;
+import com.example.bookshop.repository.CategoryRepository;
+import com.example.bookshop.service.CategoryService;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    @Autowired
+    private CategoryRepository categoryRepository;
+    @Autowired
+    private CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable)
+                .stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        return categoryMapper.toDto(categoryRepository.getReferenceById(id));
+    }
+
+    @Override
+    public CategoryDto save(CategoryDto categoryDto) {
+        return categoryMapper.toDto(categoryRepository.save(
+                        categoryMapper.toEntity(categoryDto)
+                )
+        );
```

Comment:
Let’s  fix formatting  
```suggestion
        Category category = categoryMapper.toEntity(categoryDto);
        return categoryMapper.toDto(categoryRepository.save(category));
```